# Ejemplo 5.3.1 de Díaz y Morales (2015)
<img src="https://raw.githubusercontent.com/fhernanb/fhernanb.github.io/master/my_docs/logo_unal_color.png" alt="drawing" width="200"/>

En este ejemplo se desea estudiar el siguiente conjunto de hipótesis usando un nivel de significancia de 0.05.

$H_0: \boldsymbol{\Sigma} = 
\begin{pmatrix}
4 & 3 & 2 \\
3 & 6 & 5 \\
2 & 5 & 10
\end{pmatrix}
$

$H_A: \boldsymbol{\Sigma} \neq 
\begin{pmatrix}
4 & 3 & 2 \\
3 & 6 & 5 \\
2 & 5 & 10
\end{pmatrix}
$

Suponiendo que la matriz de covarianzas muestral es

\begin{equation}
\mathbf{S} = 
\begin{pmatrix}
3.42 & 2.60 & 1.89 \\
2.60 & 8.00 & 6.51 \\
1.89 & 6.51 & 9.62
\end{pmatrix}
\end{equation}

con $n=20$ sujetos que fueron los que participaron en el estudio.

In [1]:
import numpy as np
import scipy.stats

A continuación los datos del ejemplo

In [2]:
Sigma0 = np.matrix([[4, 3, 2],
                    [3, 6, 5],
                    [2, 5, 10]])
print(Sigma0)

print("\n")

S = np.matrix([[3.42, 2.60, 1.89],
               [2.60, 8.00, 6.51],
               [1.89, 6.51, 9.62]])
print(S)

[[ 4  3  2]
 [ 3  6  5]
 [ 2  5 10]]


[[3.42 2.6  1.89]
 [2.6  8.   6.51]
 [1.89 6.51 9.62]]


## Prueba razón de verosimilitud de Rencher and Christensen (2012) page 260 expression 7.4

Esta prueba se usa para valores grandes de $n$. El estadístico está dado por

$$
\lambda^\star = (n-1) \left( \sum_{i=1}^{p} (\lambda_i - \log(\lambda_i)) - p \right)
$$

donde $\lambda_i$ son los valores propios de la matrix $\mathbf{S} \boldsymbol{\Sigma}_0^{-1}$. Cuando $H_0$ es verdadera, $\lambda^\star \sim \chi^2_{p(p+1)/2}$.

In [3]:
n = 20
p = 3

matriz_aux = np.matmul(S, Sigma0.I)
print(matriz_aux)

lambdas = np.linalg.eigvals(matriz_aux)
print(lambdas)

[[ 0.85313953 -0.01465116  0.02569767]
 [-0.57523256  1.6844186  -0.07616279]
 [-0.40918605  0.71953488  0.68406977]]
[1.62251411 0.86339991 0.73571388]


In [4]:
lambda_star = (n-1)  * (sum(lambdas - np.log(lambdas)) - p)
print(lambda_star)

3.6374040789731357


Para calcular el valor crítico con un $\alpha=0.05$ hacemos lo siguiente

In [5]:
scipy.stats.chi2.ppf(q=1-0.05, df=p*(p+1)/2)  # Valor crítico

12.591587243743977

<div class="alert alert-success">
  <strong>Resultado</strong> Como el estadístico es $\lambda^\star=3.637$, que es menor que el valor crítico $12.59$, entonces no hay evidencias suficientes para rechazar $H_0: \boldsymbol{\Sigma} = \boldsymbol{\Sigma}_0$.
</div>

## Prueba razón de verosimilitud de Rencher and Christensen (2012) page 260 expression 7.2

Esta prueba se usa para valores moderados de $n$. El estadístico está dado por

$$
\lambda^{\star}_1 = \lambda^\star \left[ 1 - \frac{1}{6(n-1)} \left( 2p + 1 - \frac{2}{p+1} \right) \right]
$$

donde $\lambda^\star$ es el valor del estadístico de la prueba anterior. Cuando $H_0$ es verdadera, $\lambda^\star \sim \chi^2_{p(p+1)/2}$.

In [6]:
lambda_star_1 = lambda_star * (1-1/(6*(n-1))*(2*p+1-2/(p+1)))
print(lambda_star_1)

3.4300082323650183


Para calcular el valor crítico con un $\alpha=0.05$ hacemos lo siguiente

In [7]:
scipy.stats.chi2.ppf(q=1-0.05, df=p*(p+1)/2)  # Valor crítico

12.591587243743977

<div class="alert alert-success">
  <strong>Resultado</strong> Como el estadístico es $\lambda^\star=3.430$, que es menor que el valor crítico $12.59$, entonces no hay evidencias suficientes para rechazar $H_0: \boldsymbol{\Sigma} = \boldsymbol{\Sigma}_0$.
</div>